.circleci/initialize.py

```diff
@@ -0,0 +1,358 @@
+
+query = gql(
+    """
+    query (
+        $name: String
+    ) {
+        current {
+            username
+            isAdmin
+        }
+        apps(
+            name: $name,
+            allApps:true,
+        ) {
+            apps {
+                name
+                owner {
+                    username
+                }
+                status {
+                    running
+                }
+                collaborators {
+                    users
+                    teams
+                }
+                metadata {
+                permissionLevel
+                }
+                linkedServices {
+                    name
+                    serviceType
+                }
+                mounts {
+                    targetDir
+                    hostDir
+                }
+                environmentVariables {
+                    name
+                    value
+                }
+            }
+        }
+    }
+    """
+)
+params = {"name": TARGET_APPNAME}
+result = client_service.execute(query, variable_values=params)
+handle_error(result, accepted_errors)
+
+if len(result["apps"]["apps"]) != 0:
+    print("Initializing review app...")
+    apps = result["apps"]["apps"]
+    apps_name = result["apps"]["apps"][0]["name"]
+    apps_owner = result["apps"]["apps"][0]["owner"]["username"]
+    apps_status = result["apps"]["apps"][0]["status"]["running"]
+    current_isAdmin = result["current"]["isAdmin"]
+    apps_collaborators = result["apps"]["apps"][0]["collaborators"]
+    apps_permissionLevels = result["apps"]["apps"][0]["metadata"]
+    apps_linkedServices = result["apps"]["apps"][0]["linkedServices"]
+    apps_mounts = result["apps"]["apps"][0]["mounts"]
+    apps_environmentVariables = result["apps"]["apps"][0]["environmentVariables"]
+
+    permissionLevels = apps_permissionLevels
+    linkedServices = zip_list_index(apps_linkedServices, "serviceType", "name")
+    mounts = zip_list_index(apps_mounts, "hostDir", "targetDir")
+    environmentVariables = zip_list_index(apps_environmentVariables, "name", "value")
+
+    query = gql(
+        """
+        mutation (
+            $appname: String
+        ) {
+            addApp(
+                name: $appname
+            ) {
+                error
+            }
+        }
+        """
+    )
+    params = {"appname": APPNAME}
+
+    result = client_user.execute(query, variable_values=params)
+    handle_error(result, accepted_errors)
+    print("  {APPNAME}".format(APPNAME=APPNAME))
+else:
+    print(result)
+    raise Exception(
+        "\nReview app not initialized"
+        + "\nApp {TARGET_APPNAME} does not exist or the user {USERNAME}".format(
+            TARGET_APPNAME=TARGET_APPNAME, USERNAME=USERNAME
+        )
+        + "does not have permission to query this app.\n"
+    )
+
+if len(linkedServices.items()) == 1:
+    print(
+        "Adding redis/postgres database...",
+    )
+elif len(linkedServices.items()) > 1:
```

Comment:
Because this is an `elif`, if there is a single service (as per line 188), this conditional won't be entered